casanovo/denovo/dataloaders.py

```
@@ -342,6 +355,288 @@ def _make_dataset(
+                        }
+
+                        if not annotated:
+                            record["charge"] = charge
+
+                        yield pl.DataFrame(record)
+
+            logging.warning(
+                f"{skipped} spectra were skipped due to missing MS1 scans for file {spectra}"
+            )
+
+    def _extract_spectra(
```

Ruff (0.15.21)

[warning] Too many branches (19 > 12)

(PLR0912)